src/Core/Domain/Order/Exception/OrderConstraintException.php

```diff
@@ -31,4 +31,8 @@
  */
 class OrderConstraintException extends OrderException
 {
+    /**
+     * @var int Is used when invalid (not string) internal note provided
+     */
+    const INVALID_INTERNAL_NOTE = 1;
```

Comment:
Careful there is already a INVALID_CUSTOMER_MESSAGE const on the 177 branch, although it is not merged yet (we have some issues with travis)

You couldn't know it of course, but you will likely have conflicts here so I suggest you use the value `2` for this one